repo2docker/utils.py

```diff
@@ -69,6 +70,22 @@ def chdir(path):
     finally:
         os.chdir(old_dir)
 
+@contextmanager
+def open_utf8convert_read(path):
+    with open(path, "rb") as f:
+        file_to_encode = f.read()
+
+    encoding_detection_result = chardet.detect(file_to_encode)
```

Comment:
Not sure it makes much difference here in terms of performance so this is more for my education/old memory: is there a way to read only a small part of the file and guess the encoding instead of having to read the whole file?

Reply:
Seems like a good idea and apparently chardet's docs have an example about [detecting encoding incrementally](https://chardet.readthedocs.io/en/latest/usage.html#advanced-usage) :confetti_ball:. 
I *think* that even though it might not impact the performance that much in *requirements.txt*'s case (I'm guessing the file is not very large in general), it might make a difference if used somewhere else, on other files :thinking:

Reply:
Took a quick look at the example in the docs that you linked to and my first impression was "woah what is going on here, no context managers? do we need code to handle errors?". Maybe we keep the "read it all at once" because it is nice and clear. Maybe we leave a comment pointing to the incremental one for later? Maybe there is a nice way to write the incremental example with modern Python? What do you think?

Reply:
Too slow in commenting/didn't see the "outdated" flag. Your code doesn't look as scary as the example so let's keep it :)